make_data_from_mat.py

```diff
@@ -0,0 +1,409 @@
+#!/usr/bin/env python
+#  $Id$
+"""
+Make gsw_check_data.c and gsw_saar_data.c from the current gsw_data_v3_0.mat,
+v3_06_11. Existing versions will be overwritten.
+
+Version recording in the mat files is completely unreliable, so we override it.
+
+One function, geo_strf_dyn_height, is in a state of flux; until it is
+completely rewritten to match the matlab v3_06_12 version, and this file
+is rerun with the corresponding matfile, the test will fail.  Using the
+pchip-based version here, the error is:
+
+  Max difference = 0.0030712207345846565, limit = 4.5372335222282345e-07
+  Max diff (rel) = 0.027299449064902257, limit = 4.0330535034756395e-06
+
+With the original algorithm, the mismatch relative to v3_06_11 is larger:
+
+  Max difference = 0.014465517624415725, limit = 4.5372335222282345e-07
+  Max diff (rel) = 0.088538110572418685, limit = 2.7770736845661537e-06
+
+
+This is a developer utility and not a part of the public distribution, but its
+end-product is.
+
+"""
+import math, os, sys
+import textwrap
+import numpy as np
+from scipy.io import loadmat
+
+
+# Edit the mat_filename as needed, but make sure there is enough path
+# info to show the matlab "version" it comes from.
+mat_filename = '../../gsw_matlab_v3_06_11/library/gsw_data_v3_0.mat'
```

Comment:
We should work with upstream to release these data in a more friendly way. Maybe a netCDF file served/hosted somewhere that would make it easy for all projects to get it.

Reply:
What is it you want in netcdf format?

Reply:
@ocefpaf is referring to all of the data in gsw_data_v3_0.mat, which should include clear version identification both inside the file and in the file name.  As it is now, there are many differing versions of that critical file, with no clarity as to what has changed from one to another, or which one is which.  This is part of a larger organizational issue of how to make it less difficult to provide gsw functionality in languages other than Matlab--and even for Matlab users, of how to track what has changed from one version to another.

Reply:
I would add, thought, that this is not an immediate priority; we are not asking you to spend time now to write a netcdf version from the matfile.  We are directly reading the matfile, which is easy in Python.

Reply:
Hah! It turns out there *is* a https://github.com/TEOS-10/GSW-datafile repo.  It was last updated Oct 29, 2015.  Here are the global attributes of the sole file in the repo:
```
		:version_date = "15th_May_2011" ;
		:version_number = "3.05" ;
		:history = "Created by Paul M. Barker on 30th October 2015 for the Gibbs-Seawater Oceanographic Toolbox" ;
		:title = "Contains the global data set of Absolute Salinity Anomaly Ratio, the global data set of Absolute Salinity Anomaly Atlas, a reference cast and check values for all of the functions" ;
```
And here they are for the version Frank Delahoyde used for his version of GSW-C:
```
		:version_date = "15th_May_2011" ;
		:version_number = "3.05.6" ;
		:history = "Created by Paul M. Barker on 8th August 2016 for the Gibbs-Seawater Oceanographic Toolbox" ;
		:title = "Contains the global data set of Absolute Salinity Anomaly Ratio, the global data set of Absolute Salinity Anomaly Atlas, a reference cast and check values for all of the functions" ;
```

Reply:
I have the data file in netcdf format, it is made during the matlab datafile making. I will put it up on the repo.

Reply:
Can (and should) all versioning be clearly matched to the matlab releases, at least so long as they are setting the standard?  So, a file name for the current latest would be, for example, "gsw_data_matlab_v3_06_12.nc".  version_number could be 3.06.12; Maybe it already is.  Maybe version_date should be the date of the matlab release, or maybe the date of the last time the actual data tables, as opposed to check values, were changed?